app/code/Magento/Backend/Controller/Adminhtml/Index/GlobalSearch.php

```diff
@@ -24,16 +28,44 @@ class GlobalSearch extends \Magento\Backend\Controller\Adminhtml\Index
     protected $_searchModules;
 
     /**
+     * modules that support preview
+     *
+     * @var array
+     */
+    private $previewModules;
+
+    /**
+     * @var ItemFactory
+     */
+    private $itemFactory;
+
+    /**
+     * @var SearchCriteriaFactory
+     */
+    private $criteriaFactory;
+
+    /**
+     * Initialize dependencies
+     *
      * @param \Magento\Backend\App\Action\Context $context
      * @param \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory
+     * @param ItemFactory $itemFactory
+     * @param SearchCriteriaFactory $criteriaFactory
      * @param array $searchModules
+     * @param array $previewModules
      */
     public function __construct(
         \Magento\Backend\App\Action\Context $context,
         \Magento\Framework\Controller\Result\JsonFactory $resultJsonFactory,
-        array $searchModules = []
+        ItemFactory $itemFactory,
+        SearchCriteriaFactory $criteriaFactory,
+        array $searchModules = [],
+        array $previewModules = []
```

Comment:
Such changes to the class constructor violate backwards compatibility policy:
* New dependencies must not be added to the middle, only as a last parameter.
* All newly added arguments must be nullable.